### PR TITLE
Hierarchical metrics for perf(no metric aggregation)

### PR DESCRIPTION
**Pull request by mmucek95**

### CODE REVIEW

File: wca/metrics.py [hunk 1]
```diff
@@ -373,6 +403,30 @@ def create_metric_with_metadata(name, value, labels=None, granularity=None):
 Measurements = Dict[MetricName, MetricValue]
 
 
+def _sum_leveled_metrics(aggregated_metric, new_metric, max_depth, depth=0):
```
Comment (ppalucki) [inline wca/metrics.py]: please add a short docstring to express intent and some unittests :)

File: wca/metrics.py [hunk 2]
```diff
@@ -385,20 +439,30 @@ def merge_measurements(measurements_list: List[Measurements]) -> \
...
 
-        summed_metrics[metric_name] = operation(
-            [measurements[metric_name] for measurements in measurements_list
-             if metric_name in measurements])
+        if metric_name not in METRICS_LEVELS:
+            try:
+                summed_metrics[metric_name] = operation(
+                    [measurements[metric_name] for measurements in measurements_list
+                     if metric_name in measurements])
+            except TypeError:
```
Comment (ppalucki) [inline wca/metrics.py]: why TypeError ? what we're trying to catch here ?

File: wca/metrics.py [hunk 3]
```diff
@@ -434,11 +498,34 @@ def available(*names):
         def delta(*names):
             if not available(*names):
                 return 0
-            if len(names) == 1:
-                name = names[0]
-                return measurements[name] - self._prev_measurements[name]
-            else:
-                return [measurements[name] - self._prev_measurements[name] for name in names]
+
+            def substract_leveled_metrics(
```
Comment (ppalucki) [inline wca/metrics.py]: if possible, move outside and add some short docstring and unittests

File: wca/storage.py [hunk 4]
```diff
@@ -389,6 +389,31 @@ def callback_on_delivery(self, err, msg) -> None:
             log.log(logger.TRACE,
                     'KafkaStorage succeeded to send message; message: {}'.format(msg))
 
+    @staticmethod
+    def divide_message(msg):
```
Comment (ppalucki) [inline wca/storage.py]: ditto :) docstring and unittest (even very simple one)

docstring should explain why we need this (the limit of Kafka server in accepting >1Mb messages)

File: wca/metrics.py [hunk 5]
```diff
@@ -385,24 +430,81 @@ def merge_measurements(measurements_list: List[Measurements]) -> \
...
+                     if metric_name in measurements])
+            except TypeError:
+                log.exception("{} seems to be hierarchical but metric levels are "
+                              "not specified.".format(metric_name))
+                raise
+        else:
+            max_depth = len(METRICS_LEVELS[metric_name])
+            summed = dict()
+            for measurements in measurements_list:
+                summed = _list_leveled_metrics(summed, measurements[metric_name], max_depth)
```
Comment (felidadae) [inline wca/metrics.py]: lacking condition if metric_name in measurements

File: wca/metrics.py (hunk 1 above)
Comment (ppalucki) [reply wca/metrics.py]: done

File: wca/metrics.py (hunk 3 above)
Comment (ppalucki) [reply wca/metrics.py]: done

**Comment by ppalucki** [comment]
superseded by #321

**Pull request closed without merging**